### ISSUE TITLE
assertion rewrite broken in 3.5

### ISSUE THREAD

**Issue by llllllllll** [pytest-dev/pytest#996]
Tried using the pytest runner with 3.5 and it looks like the `ast.Call` constructor changed
```
../../../.virtualenvs/cotoolz/lib/python3.5/site-packages/py/_path/local.py:650: in pyimport
    __import__(modname)
<frozen importlib._bootstrap>:969: in _find_and_load
    ???
<frozen importlib._bootstrap>:954: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:892: in _find_spec
    ???
<frozen importlib._bootstrap>:873: in _find_spec_legacy
    ???
../../../.virtualenvs/cotoolz/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:137: in find_module
    source_stat, co = _rewrite_test(state, fn_pypath)
../../../.virtualenvs/cotoolz/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:278: in _rewrite_test
    rewrite_asserts(tree)
../../../.virtualenvs/cotoolz/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:336: in rewrite_asserts
    AssertionRewriter().run(mod)
../../../.virtualenvs/cotoolz/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:557: in run
    new.extend(self.visit(child))
/usr/local/lib/python3.5/ast.py:245: in visit
    return visitor(node)
../../../.virtualenvs/cotoolz/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:665: in
 visit_Assert
    top_condition, explanation = self.visit(assert_.test)
/usr/local/lib/python3.5/ast.py:245: in visit
    return visitor(node)
../../../.virtualenvs/cotoolz/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:796: in visit_Compare
    left_res, left_expl = self.visit(comp.left)
/usr/local/lib/python3.5/ast.py:245: in visit
    return visitor(node)
../../../.virtualenvs/cotoolz/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:757: in visit_Call
    new_func, func_expl = self.visit(call.func)
/usr/local/lib/python3.5/ast.py:245: in visit
    return visitor(node)
../../../.virtualenvs/cotoolz/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:700: in visit_Name
    locs = ast.Call(self.builtin("locals"), [], [], None, None)
E   TypeError: Call constructor takes either 0 or 3 positional arguments
```

**Comment by The-Compiler** [comment]
This was already fixed in #801 - there hopefully will be a 2.8 release with that in soon. As a workaround, you can use `--assert=plain`.

**Comment by llllllllll** [comment]
sorry about this.
